stochastic_physics/stochastic_physics_wrapper.F90

```diff
@@ -386,18 +393,24 @@ subroutine stochastic_physics_wrapper (GFS_Control, GFS_Data, Atm_block, ierr)
 
        if(GFS_Control%ca_sgs)then
          do nb=1,nblks
-             sst        (nb,1:GFS_Control%blksz(nb)) = GFS_Data(nb)%Sfcprop%tsfco(:)
-             lmsk       (nb,1:GFS_Control%blksz(nb)) = GFS_Data(nb)%Sfcprop%slmsk(:)
-             lake       (nb,1:GFS_Control%blksz(nb)) = GFS_Data(nb)%Sfcprop%lakefrac(:)
-             uwind      (nb,1:GFS_Control%blksz(nb),:) = GFS_Data(nb)%Statein%ugrs(:,:)
-             vwind      (nb,1:GFS_Control%blksz(nb),:) =  GFS_Data(nb)%Statein%vgrs(:,:)
-             height     (nb,1:GFS_Control%blksz(nb),:) =  GFS_Data(nb)%Statein%phil(:,:)
-             dx         (nb,1:GFS_Control%blksz(nb)) = GFS_Data(nb)%Grid%dx(:)
-             condition  (nb,1:GFS_Control%blksz(nb)) = GFS_Data(nb)%Coupling%condition(:)
-             ca_deep_cpl(nb,1:GFS_Control%blksz(nb)) = GFS_Data(nb)%Coupling%ca_deep(:)
-             ca_turb_cpl(nb,1:GFS_Control%blksz(nb)) = GFS_Data(nb)%Coupling%ca_turb(:)
-             ca_shal_cpl(nb,1:GFS_Control%blksz(nb)) = GFS_Data(nb)%Coupling%ca_shal(:)
+             ixs = GFS_control%chunk_begin(nb)
+             ixe = GFS_control%chunk_end(nb)
+             sst        (nb,1:GFS_Control%blksz(nb)) = GFS_Sfcprop%tsfco(ixs:ixe)
+             lmsk       (nb,1:GFS_Control%blksz(nb)) = GFS_Sfcprop%slmsk(ixs:ixe)
+             lake       (nb,1:GFS_Control%blksz(nb)) = GFS_Sfcprop%lakefrac(ixs:ixe)
+             ! DH* note - don't think we need the transfer_field logic, just use ixs:ixe?
+             !uwind      (nb,1:GFS_Control%blksz(nb),:) = GFS_Data(nb)%Statein%ugrs(:,:)
+             call transfer_field_to_stochastics_3d(GFS_Control%blksz, GFS_Statein%ugrs, uwind)
```

Comment:
@climbfuji  The slowdown was happening the stochastic physics (GFS Step Setup (2))
![Screen Shot 2024-07-11 at 1 19 41 PM](https://github.com/NOAA-EMC/fv3atm/assets/13949033/187ade36-6564-46b3-80c8-d147a6c7d14e)

Here we are in a loop over blocks.
transfer_field_to_stochastics_3d loops over layers and calls transfer_field_to_stochastics, which has a loop over the blocks. 
I think we need to move the calls to transfer_field_to_stochastics_3d from this loop to just after it?
(I'm testing this now)

Reply:
Awww that's an awesome find! Thanks Dustin, I think you are totally right about this.

Reply:
Some exciting news!
After moving the call to transfer_field_to_stochastics_3d outside of the block loop:
![Screen Shot 2024-07-11 at 9 39 16 PM](https://github.com/user-attachments/assets/17a3b79a-6d1b-4193-9baa-98f0fd43d781)

w/o this change:
![Screenshot 2024-07-11 at 9 02 47 AM](https://github.com/user-attachments/assets/e5a9d332-3621-44ba-a274-e6bffe185bf2)

and develop branch:
![Screenshot 2024-07-11 at 9 02 06 AM](https://github.com/user-attachments/assets/90c40df4-81cf-45cc-b42a-f251a3021919)

Reply:
This is awesome! Looks like from your tests (acknowledging that these may be single runs) that there are performance gains in `GFS Step Setup` (and maybe smaller ones in `Dynamics get state`, whereas `GFS Physics` is slightly slower? So overall maybe just about equal (3733 develop versus 3508 this code with your bug fix). And what happens if you change the blocksize to `64` instead of `32`? @DusanJovic-NOAA saw performance improvements for that, too.

Exciting!

If the code is still b4b in full rt.sh runs, can you submit the updates to my branch please (or create a PR for my branch)? Thanks very much!

Reply:
Yeah, it's super exciting to see a speedup! Even if it's just one simulation.
Checking for b4b now (ATM control_p8). I will test the blocksize change, but I also want to look into the physics slowdown as well. 
I will open PRs into your branch once I confirm everything is working as expected.

Reply:
@DomHeinzeller b4b confirmed. Running the c1152 (FV3_GFS_v17_p8_ugwpv1) with blocksize=64 now.

Reply:
Thanks! In the meanwhile, I updated my code from the authoriative branches, and unfortunately got one failure for `hafs_regional_storm_following_1nest_atm_ocn_debug_intel` on Hercules :-(

See https://github.com/ufs-community/ufs-weather-model/pull/2183#issuecomment-2226692259

I'll look into this the next week(s) unless someone tells me that this is a known problem.

Reply:
Oh, maybe it is a known problem (the one that was mentioned at the code managers meeting today)? https://github.com/ufs-community/ufs-weather-model/issues/2360